src/utils/gasPriceManager.ts

```diff
@@ -0,0 +1,450 @@
+import {
+    gasStationResult,
+    type GasPriceParameters,
+    RpcError
+} from "@entrypoint-0.7/types"
+import { parseGwei, type Chain, type PublicClient } from "viem"
+import { type Logger, maxBigInt, minBigInt } from "@alto/utils"
+import * as chains from "viem/chains"
+import * as sentry from "@sentry/node"
+
+export interface InterfaceGasPriceManager {
+    getGasPrice(): Promise<GasPriceParameters>
+    validateGasPrice(gasPrice: GasPriceParameters): void
+}
+
+enum ChainId {
+    Goerli = 5,
+    Polygon = 137,
+    Mumbai = 80001,
+    LineaTestnet = 59140,
+    Linea = 59144
+}
+
+const MIN_POLYGON_GAS_PRICE = parseGwei("31")
+const MIN_MUMBAI_GAS_PRICE = parseGwei("1")
+
+function getGasStationUrl(chainId: ChainId.Polygon | ChainId.Mumbai): string {
+    switch (chainId) {
+        case ChainId.Polygon:
+            return "https://gasstation.polygon.technology/v2"
+        case ChainId.Mumbai:
+            return "https://gasstation-testnet.polygon.technology/v2"
+    }
+}
+
+export class GasPriceManager implements InterfaceGasPriceManager {
+    private chain: Chain
+    private publicClient: PublicClient
+    private noEip1559Support: boolean
+    private logger: Logger
+    private queueMaxFeePerGas: { timestamp: number; maxFeePerGas: bigint }[] =
+        [] // Store pairs of [price, timestamp]
+    private queueMaxPriorityFeePerGas: {
+        timestamp: number
+        maxPriorityFeePerGas: bigint
+    }[] = [] // Store pairs of [price, timestamp]
+    private maxQueueSize
```

Comment:
how big are we looking to set this?

Reply:
10 (one for each second) - this is configurable